optuna/samplers/_gp/sampler.py

```diff
@@ -105,6 +113,17 @@ def infer_relative_search_space(
 
         return search_space
 
+    def _optimize_acqf(
+        self,
+        acqf_params: "acqf.AcquisitionFunctionParams",
+    ) -> np.ndarray:
+        normalized_params, _ = optim_sample.optimize_acqf_sample(
```

Comment:
We can leave it for now, but let me add some comments because we should delete this method unless we get its strong demand from users.
```suggestion
        # This method is supposed to be overridden by advanced users or developers
        # to try different optimization strategies.
        # TODO(nabenabe0928): Remove this method once our optimization strategy is fixed.
        normalized_params, _ = optim_sample.optimize_acqf_sample(
```

Reply:
Do you have a strong reason that this function should be removed?
People have different budget requirements. Some people can put up with 1 hour optimizing their acquisition function, and others cannot stand 1 second. I don't think we can ever make a silver bullet optimization method that works for everyone.

Reply:
```suggestion
        # Advanced users can override this method to change the optimization algorithm.
        # However, we do not make any effort to keep backward compatibility between versions.
        normalized_params, _ = optim_sample.optimize_acqf_sample(
```
How about this comment?

Reply:
Sounds great!